### ISSUE TITLE
golem_messages.exceptions.FieldError: Port not in range 1, 2**16-1 [port:0]

### ISSUE THREAD

**Issue by ederenn** [golemfactory/clay#3817]
#### Encounered on:

Linux, mainnet
GOLEM Version: 0.18.3+dev739.gb7573ca
Protocol Version: 31
golem_messages Version: 2.23.0

around 10 minutes after starting the node, no upgrade, no additional changes, no new tasks started, just starting. No problems with network at this time.

```
2019-01-24 12:13:13 CRITICAL twisted                             Unhandled Error
Traceback (most recent call last):
  File "/home/buildbot-worker/worker/buildpackage_linux/build/.venv/lib/python3.6/site-packages/twisted/python/log.py", line 103, in callWithLogger
    
  File "/home/buildbot-worker/worker/buildpackage_linux/build/.venv/lib/python3.6/site-packages/twisted/python/log.py", line 86, in callWithContext
    
  File "/home/buildbot-worker/worker/buildpackage_linux/build/.venv/lib/python3.6/site-packages/twisted/python/context.py", line 122, in callWithContext
    
  File "/home/buildbot-worker/worker/buildpackage_linux/build/.venv/lib/python3.6/site-packages/twisted/python/context.py", line 85, in callWithContext
    
--- <exception caught here> ---
  File "/home/buildbot-worker/worker/buildpackage_linux/build/.venv/lib/python3.6/site-packages/twisted/internet/posixbase.py", line 614, in _doReadOrWrite
    
  File "/home/buildbot-worker/worker/buildpackage_linux/build/.venv/lib/python3.6/site-packages/twisted/internet/tcp.py", line 205, in doRead
    
  File "/home/buildbot-worker/worker/buildpackage_linux/build/.venv/lib/python3.6/site-packages/twisted/internet/tcp.py", line 211, in _dataReceived
    
  File "/home/buildbot-worker/worker/buildpackage_linux/build/golem/network/transport/tcpnetwork.py", line 335, in dataReceived
    
  File "/home/buildbot-worker/worker/buildpackage_linux/build/golem/network/transport/tcpnetwork.py", line 361, in _interpret
    
  File "/home/buildbot-worker/worker/buildpackage_linux/build/golem/network/p2p/peersession.py", line 119, in interpret
    
  File "/home/buildbot-worker/worker/buildpackage_linux/build/golem/network/transport/session.py", line 75, in interpret
    
  File "/home/buildbot-worker/worker/buildpackage_linux/build/golem/network/p2p/peersession.py", line 332, in _react_to_get_peers
    
  File "/home/buildbot-worker/worker/buildpackage_linux/build/golem/network/p2p/peersession.py", line 506, in _send_peers
    
  File "/home/buildbot-worker/worker/buildpackage_linux/build/golem/network/p2p/p2pservice.py", line 652, in find_node
    
  File "/home/buildbot-worker/worker/buildpackage_linux/build/golem/network/p2p/p2pservice.py", line 652, in <listcomp>
    
  File "/home/buildbot-worker/worker/buildpackage_linux/build/golem/network/p2p/p2pservice.py", line 650, in _mapper_session
    
  File "/home/buildbot-worker/worker/buildpackage_linux/build/.venv/lib/python3.6/site-packages/golem_messages/datastructures/__init__.py", line 53, in __init__
    
  File "/home/buildbot-worker/worker/buildpackage_linux/build/.venv/lib/python3.6/site-packages/golem_messages/datastructures/__init__.py", line 30, in __init__
    
  File "/home/buildbot-worker/worker/buildpackage_linux/build/.venv/lib/python3.6/site-packages/golem_messages/datastructures/p2p.py", line 112, in __setitem__
    
  File "/home/buildbot-worker/worker/buildpackage_linux/build/.venv/lib/python3.6/site-packages/golem_messages/datastructures/__init__.py", line 82, in __setitem__
    
  File "/home/buildbot-worker/worker/buildpackage_linux/build/.venv/lib/python3.6/site-packages/golem_messages/validators.py", line 41, in validate_port
    
golem_messages.exceptions.FieldError: Port not in range 1, 2**16-1 [port:0]
```

**Comment by ederenn** [comment]
repeated in 

```
GOLEM Version: 0.18.3+dev762.gdfaf628
Protocol Version: 31
 golem_messages Version: 2.24.0
```

Connected nodes start to disconnect randomly. In a while the are establishing connection again

```
019-01-25 13:10:35 INFO     golem.network.p2p.p2pservice        Adding peer 'vm win 19'(b185150d..a92f0362), key id difficulty: 16
2019-01-25 13:10:35 CRITICAL twisted                             Unhandled Error
Traceback (most recent call last):
  File "/home/buildbot-worker/worker/buildpackage_linux/build/.venv/lib/python3.6/site-packages/twisted/python/log.py", line 103, in callWithLogger
    
  File "/home/buildbot-worker/worker/buildpackage_linux/build/.venv/lib/python3.6/site-packages/twisted/python/log.py", line 86, in callWithContext
    
  File "/home/buildbot-worker/worker/buildpackage_linux/build/.venv/lib/python3.6/site-packages/twisted/python/context.py", line 122, in callWithContext
    
  File "/home/buildbot-worker/worker/buildpackage_linux/build/.venv/lib/python3.6/site-packages/twisted/python/context.py", line 85, in callWithContext
    
--- <exception caught here> ---
  File "/home/buildbot-worker/worker/buildpackage_linux/build/.venv/lib/python3.6/site-packages/twisted/internet/posixbase.py", line 614, in _doReadOrWrite
    
  File "/home/buildbot-worker/worker/buildpackage_linux/build/.venv/lib/python3.6/site-packages/twisted/internet/tcp.py", line 205, in doRead
    
  File "/home/buildbot-worker/worker/buildpackage_linux/build/.venv/lib/python3.6/site-packages/twisted/internet/tcp.py", line 211, in _dataReceived
    
  File "/home/buildbot-worker/worker/buildpackage_linux/build/.venv/lib/python3.6/site-packages/twisted/internet/endpoints.py", line 125, in dataReceived
    
  File "/home/buildbot-worker/worker/buildpackage_linux/build/golem/network/transport/tcpnetwork.py", line 335, in dataReceived
    
  File "/home/buildbot-worker/worker/buildpackage_linux/build/golem/network/transport/tcpnetwork.py", line 361, in _interpret
    
  File "/home/buildbot-worker/worker/buildpackage_linux/build/golem/network/p2p/peersession.py", line 119, in interpret
    
  File "/home/buildbot-worker/worker/buildpackage_linux/build/golem/network/transport/session.py", line 75, in interpret
    
  File "/home/buildbot-worker/worker/buildpackage_linux/build/golem/network/p2p/peersession.py", line 332, in _react_to_get_peers
    
  File "/home/buildbot-worker/worker/buildpackage_linux/build/golem/network/p2p/peersession.py", line 508, in _send_peers
    
  File "/home/buildbot-worker/worker/buildpackage_linux/build/golem/network/p2p/p2pservice.py", line 652, in find_node
    
  File "/home/buildbot-worker/worker/buildpackage_linux/build/golem/network/p2p/p2pservice.py", line 652, in <listcomp>
    
  File "/home/buildbot-worker/worker/buildpackage_linux/build/golem/network/p2p/p2pservice.py", line 650, in _mapper_session
    
  File "/home/buildbot-worker/worker/buildpackage_linux/build/.venv/lib/python3.6/site-packages/golem_messages/datastructures/__init__.py", line 53, in __init__
    
  File "/home/buildbot-worker/worker/buildpackage_linux/build/.venv/lib/python3.6/site-packages/golem_messages/datastructures/__init__.py", line 30, in __init__
    
  File "/home/buildbot-worker/worker/buildpackage_linux/build/.venv/lib/python3.6/site-packages/golem_messages/datastructures/p2p.py", line 112, in __setitem__
    
  File "/home/buildbot-worker/worker/buildpackage_linux/build/.venv/lib/python3.6/site-packages/golem_messages/datastructures/__init__.py", line 82, in __setitem__
    
  File "/home/buildbot-worker/worker/buildpackage_linux/build/.venv/lib/python3.6/site-packages/golem_messages/validators.py", line 41, in validate_port
    
golem_messages.exceptions.FieldError: Port not in range 1, 2**16-1 [port:0]

2019-01-25 13:10:39 INFO     golem.network.p2p.peersession       Starting peer session '10.30.11.251':46101
2019-01-25 13:10:39 WARNING  golem.network.p2p.peersession       PEER DUPLICATED: 'vm win 19' '10.30.11.251' : 53060 AND 'vm win 19' : 46101
2019-01-25 13:10:39 INFO     golem.network.transport.session     Disconnecting '10.30.11.251':46101 reason: <REASON.DuplicatePeers: 'duplicate_peers'>
2019-01-25 13:10:44 CRITICAL twisted                             Unhandled Error
Traceback (most recent call last):
  File "/home/buildbot-worker/worker/buildpackage_linux/build/.venv/lib/python3.6/site-packages/twisted/python/log.py", line 103, in callWithLogger
    
  File "/home/buildbot-worker/worker/buildpackage_linux/build/.venv/lib/python3.6/site-packages/twisted/python/log.py", line 86, in callWithContext
    
  File "/home/buildbot-worker/worker/buildpackage_linux/build/.venv/lib/python3.6/site-packages/twisted/python/context.py", line 122, in callWithContext
    
  File "/home/buildbot-worker/worker/buildpackage_linux/build/.venv/lib/python3.6/site-packages/twisted/python/context.py", line 85, in callWithContext
    
--- <exception caught here> ---
  File "/home/buildbot-worker/worker/buildpackage_linux/build/.venv/lib/python3.6/site-packages/twisted/internet/posixbase.py", line 614, in _doReadOrWrite
    
  File "/home/buildbot-worker/worker/buildpackage_linux/build/.venv/lib/python3.6/site-packages/twisted/internet/tcp.py", line 205, in doRead
    
  File "/home/buildbot-worker/worker/buildpackage_linux/build/.venv/lib/python3.6/site-packages/twisted/internet/tcp.py", line 211, in _dataReceived
    
  File "/home/buildbot-worker/worker/buildpackage_linux/build/.venv/lib/python3.6/site-packages/twisted/internet/endpoints.py", line 125, in dataReceived
    
  File "/home/buildbot-worker/worker/buildpackage_linux/build/golem/network/transport/tcpnetwork.py", line 335, in dataReceived
    
  File "/home/buildbot-worker/worker/buildpackage_linux/build/golem/network/transport/tcpnetwork.py", line 361, in _interpret
    
  File "/home/buildbot-worker/worker/buildpackage_linux/build/golem/network/p2p/peersession.py", line 119, in interpret
    
  File "/home/buildbot-worker/worker/buildpackage_linux/build/golem/network/transport/session.py", line 75, in interpret
    
  File "/home/buildbot-worker/worker/buildpackage_linux/build/golem/network/p2p/peersession.py", line 421, in _react_to_find_node
    
  File "/home/buildbot-worker/worker/buildpackage_linux/build/golem/network/p2p/peersession.py", line 508, in _send_peers
    
  File "/home/buildbot-worker/worker/buildpackage_linux/build/golem/network/p2p/p2pservice.py", line 663, in find_node
    
  File "/home/buildbot-worker/worker/buildpackage_linux/build/golem/network/p2p/p2pservice.py", line 663, in <listcomp>
    
  File "/home/buildbot-worker/worker/buildpackage_linux/build/golem/network/p2p/p2pservice.py", line 660, in _mapper
    
  File "/home/buildbot-worker/worker/buildpackage_linux/build/.venv/lib/python3.6/site-packages/golem_messages/datastructures/__init__.py", line 53, in __init__
    
  File "/home/buildbot-worker/worker/buildpackage_linux/build/.venv/lib/python3.6/site-packages/golem_messages/datastructures/__init__.py", line 30, in __init__
    
  File "/home/buildbot-worker/worker/buildpackage_linux/build/.venv/lib/python3.6/site-packages/golem_messages/datastructures/p2p.py", line 112, in __setitem__
    
  File "/home/buildbot-worker/worker/buildpackage_linux/build/.venv/lib/python3.6/site-packages/golem_messages/datastructures/__init__.py", line 82, in __setitem__
    
  File "/home/buildbot-worker/worker/buildpackage_linux/build/.venv/lib/python3.6/site-packages/golem_messages/validators.py", line 36, in validate_port
    
  File "/home/buildbot-worker/worker/buildpackage_linux/build/.venv/lib/python3.6/site-packages/golem_messages/validators.py", line 31, in validate_integer
    
golem_messages.exceptions.FieldError: Should be an integer [port:None]
```

**Comment by shadeofblue** [comment]
@ederenn please confirm if this is still an issue

**Comment by ederenn** [comment]
yes it is

GOLEM Version: 0.18.3+dev771.gdeb8b58
Protocol Version: 31
golem_messages Version: 2.24.1
system: Linux, mainnet

golem runs normally but in logs repeats several times

```
2019-01-31 13:51:11 CRITICAL twisted                             Unhandled Error
Traceback (most recent call last):
  File "/home/buildbot-worker/worker/buildpackage_linux/build/.venv/lib/python3.6/site-packages/twisted/python/log.py", line 103, in callWithLogger
    
  File "/home/buildbot-worker/worker/buildpackage_linux/build/.venv/lib/python3.6/site-packages/twisted/python/log.py", line 86, in callWithContext
    
  File "/home/buildbot-worker/worker/buildpackage_linux/build/.venv/lib/python3.6/site-packages/twisted/python/context.py", line 122, in callWithContext
    
  File "/home/buildbot-worker/worker/buildpackage_linux/build/.venv/lib/python3.6/site-packages/twisted/python/context.py", line 85, in callWithContext
    
--- <exception caught here> ---
  File "/home/buildbot-worker/worker/buildpackage_linux/build/.venv/lib/python3.6/site-packages/twisted/internet/posixbase.py", line 614, in _doReadOrWrite
    
  File "/home/buildbot-worker/worker/buildpackage_linux/build/.venv/lib/python3.6/site-packages/twisted/internet/tcp.py", line 205, in doRead
    
  File "/home/buildbot-worker/worker/buildpackage_linux/build/.venv/lib/python3.6/site-packages/twisted/internet/tcp.py", line 211, in _dataReceived
    
  File "/home/buildbot-worker/worker/buildpackage_linux/build/golem/network/transport/tcpnetwork.py", line 335, in dataReceived
    
  File "/home/buildbot-worker/worker/buildpackage_linux/build/golem/network/transport/tcpnetwork.py", line 361, in _interpret
    
  File "/home/buildbot-worker/worker/buildpackage_linux/build/golem/network/p2p/peersession.py", line 119, in interpret
    
  File "/home/buildbot-worker/worker/buildpackage_linux/build/golem/network/transport/session.py", line 75, in interpret
    
  File "/home/buildbot-worker/worker/buildpackage_linux/build/golem/network/p2p/peersession.py", line 418, in _react_to_find_node
    
  File "/home/buildbot-worker/worker/buildpackage_linux/build/golem/network/p2p/peersession.py", line 505, in _send_peers
    
  File "/home/buildbot-worker/worker/buildpackage_linux/build/golem/network/p2p/p2pservice.py", line 674, in find_node
    
  File "/home/buildbot-worker/worker/buildpackage_linux/build/golem/network/p2p/p2pservice.py", line 674, in <listcomp>
    
  File "/home/buildbot-worker/worker/buildpackage_linux/build/golem/network/p2p/p2pservice.py", line 671, in _mapper
    
  File "/home/buildbot-worker/worker/buildpackage_linux/build/.venv/lib/python3.6/site-packages/golem_messages/datastructures/__init__.py", line 53, in __init__
    
  File "/home/buildbot-worker/worker/buildpackage_linux/build/.venv/lib/python3.6/site-packages/golem_messages/datastructures/__init__.py", line 30, in __init__
    
  File "/home/buildbot-worker/worker/buildpackage_linux/build/.venv/lib/python3.6/site-packages/golem_messages/datastructures/p2p.py", line 112, in __setitem__
    
  File "/home/buildbot-worker/worker/buildpackage_linux/build/.venv/lib/python3.6/site-packages/golem_messages/datastructures/__init__.py", line 82, in __setitem__
    
  File "/home/buildbot-worker/worker/buildpackage_linux/build/.venv/lib/python3.6/site-packages/golem_messages/validators.py", line 36, in validate_port
    
  File "/home/buildbot-worker/worker/buildpackage_linux/build/.venv/lib/python3.6/site-packages/golem_messages/validators.py", line 31, in validate_integer
    
golem_messages.exceptions.FieldError: Should be an integer [port:None]
```

**Comment by dybi** [comment]
Closing as https://github.com/golemfactory/golem/pull/3847 has been merged (and for whatever reason hasn't automatically closed the issue).

**Comment by ederenn** [comment]
repeated on Mac, 0.19.1 released version, direct upgrade from 0.19.0


```2019-04-10 13:59:13 ERROR    twisted                             FieldError: Port not in range 1, 2**16-1 [pub_port:0]: Traceback (most recent call last):
  File "/Users/buildbot-worker/worker/buildpackage_macOS/build/.venv/lib/python3.6/site-packages/autobahn/twisted/websocket.py", line 162, in _onMessage
    
  File "/Users/buildbot-worker/worker/buildpackage_macOS/build/.venv/lib/python3.6/site-packages/autobahn/wamp/websocket.py", line 95, in onMessage
    
  File "/Users/buildbot-worker/worker/buildpackage_macOS/build/.venv/lib/python3.6/site-packages/autobahn/wamp/protocol.py", line 895, in onMessage
    
  File "/Users/buildbot-worker/worker/buildpackage_macOS/build/.venv/lib/python3.6/site-packages/txaio/tx.py", line 417, in as_future
    
--- <exception caught here> ---
  File "/Users/buildbot-worker/worker/buildpackage_macOS/build/.venv/lib/python3.6/site-packages/twisted/internet/defer.py", line 150, in maybeDeferred
    
  File "/Users/buildbot-worker/worker/buildpackage_macOS/build/golem/client.py", line 987, in get_payments_list
    
  File "/Users/buildbot-worker/worker/buildpackage_macOS/build/golem/ethereum/transactionsystem.py", line 339, in get_payments_list
    
  File "/Users/buildbot-worker/worker/buildpackage_macOS/build/golem/ethereum/paymentskeeper.py", line 115, in get_list_of_all_payments
    
  File "/Users/buildbot-worker/worker/buildpackage_macOS/build/golem/ethereum/paymentskeeper.py", line 105, in <listcomp>
    
  File "/Users/buildbot-worker/worker/buildpackage_macOS/build/.venv/lib/python3.6/site-packages/peewee.py", line 2289, in next
    
  File "/Users/buildbot-worker/worker/buildpackage_macOS/build/.venv/lib/python3.6/site-packages/peewee.py", line 2349, in iterate
    
  File "/Users/buildbot-worker/worker/buildpackage_macOS/build/.venv/lib/python3.6/site-packages/peewee.py", line 2432, in process_row
    
  File "/Users/buildbot-worker/worker/buildpackage_macOS/build/golem/model.py", line 186, in python_value
    
  File "/Users/buildbot-worker/worker/buildpackage_macOS/build/golem/model.py", line 231, in from_dict
    
  File "/Users/buildbot-worker/worker/buildpackage_macOS/build/.venv/lib/python3.6/site-packages/golem_messages/datastructures/__init__.py", line 111, in __init__
    
  File "/Users/buildbot-worker/worker/buildpackage_macOS/build/.venv/lib/python3.6/site-packages/golem_messages/datastructures/__init__.py", line 131, in load_slots
    
  File "/Users/buildbot-worker/worker/buildpackage_macOS/build/.venv/lib/python3.6/site-packages/golem_messages/datastructures/__init__.py", line 148, in deserialize_slot
    
  File "/Users/buildbot-worker/worker/buildpackage_macOS/build/.venv/lib/python3.6/site-packages/golem_messages/validators.py", line 41, in validate_port
    
golem_messages.exceptions.FieldError: Port not in range 1, 2**16-1 [pub_port:0]
```

**Comment by shadeofblue** [comment]
@ederenn I believe this is a different issue
